Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py

```diff
@@ -0,0 +1,1778 @@
+    }
+    return hr, ec, response
+
+
+def get_user_alerts_command():
+    """Retrieves a collection of Alerts related to a given user ID.
+    Returns:
+        (str, dict, dict). Human readable, context, raw response
+    """
+    headers = ['ID', 'Title', 'Description', 'IncidentID', 'Severity', 'Status', 'Classification', 'Category',
+               'ThreatFamilyName', 'MachineID']
+    username = demisto.args().get('username')
+    response = get_user_alerts_request(username)
+    alerts_list = []
+    for alert in response['value']:
+        alerts_list.append(get_alert_data(alert['id']))
+    hr = tableToMarkdown(f'User {username} related alerts Info:', alerts_list, headers=headers)
+    context_output = {
+        'Username': username,
+        'Alerts': alerts_list
+    }
+    ec = {
+        'MicrosoftATP.UserAlert(val.Username === obj.Username)': context_output
+    }
+    return hr, ec, response
+
+
+def get_user_machine_command():
+    """Retrieves a collection of machines related to a given user ID.
+    Returns:
+        (str, dict, dict). Human readable, context, raw response
+    """
+    headers = ['ID', 'ComputerDNSName', 'OSPlatform', 'LastIPAddress', 'LastExternalIPAddress', 'HealthStatus',
+               'RiskScore', 'ExposureLevel']
+    username = demisto.args().get('username')
+    response = get_user_machines_request(username)
+    machines_list = []
+    for machine in response['value']:
+        machines_list.append(get_machine_data(machine['id']))
+    hr = tableToMarkdown(f'Machines that are related to user {username}:', machines_list, headers=headers)
+    context_output = {
+        'Username': username,
+        'Machines': machines_list
+    }
+    ec = {
+        'MicrosoftATP.UserMachine(val.Username === obj.Username)': context_output
+    }
+    return hr, ec, response
+
+
+def add_remove_machine_tag_command():
+    """Adds or remove tag to a specific Machine.
+    Returns:
+        (str, dict, dict). Human readable, context, raw response
+    """
+    headers = ['ID', 'ComputerDNSName', 'OSPlatform', 'LastIpAddress', 'LastExternalIPAddress', 'HealthStatus',
+               'RiskScore', 'ExposureLevel']
+    machine_id = demisto.args().get('machine_id')
+    action = demisto.args().get('action')
+    tag = demisto.args().get('tag')
+    response = add_remove_machine_tag_request(machine_id, action, tag)
+    machine_data = get_machine_data(machine_id)
+    hr = tableToMarkdown(f'Succeed to {action} tag to {machine_id}:', machine_data, headers=headers)
+    ec = {
+        'MicrosoftATP.Machine(val.ID === obj.ID)': machine_data
+    }
+    return hr, ec, response
+
+
+def fetch_incidents():
+    last_run = demisto.getLastRun()
+
+    if last_run and last_run['last_alert_fetched_time']:
+        last_alert_fetched_time = datetime.strptime(last_run['last_alert_fetched_time'], '%Y-%m-%dT%H:%M:%S.%f')
+    else:
+        last_alert_fetched_time = datetime.now() - timedelta(days=300)
```

Comment:
300 should be a const, and should probably be a parameter and not a magic number